SPEC.md

```diff
@@ -534,7 +535,12 @@ The following primitive types exist in WDL:
 * A `File` represents a file (or file-like object).
     * A `File` declaration can have a string value indicating a relative or absolute path on the local file system.
     * Within a WDL file, literal values for files may only be local (relative or absolute) paths.
+    * The path assigned to a `File` is not required to be valid unless and until it is accessed.
+      * To read from a file, it must exist and be assigned appropriate permissions.
+      * To write to a file, the parent directory must be assigned appropriate permissions.
```

Comment:
WDL `File` objects are snapshots, right? Any WDL function that writes a `File` returns the written `File` and doesn't take a `File` as input to overwrite.

Or does this mean within a task, if you substitute a `File`'s name into a command that tries to overwrite it, the parent directory that the `File` initially came from should be writable?

Reply:
It means the directory in the execution environment where the file is to be written must have writable permissions.

Reply:
For example, if I define `File outfile = "outfile"` and then in the command do `printf "hello" > ~{outfile}`, the directory where the execution engine resolves `outfile` (presumably, the working directory) must be writable.

Reply:
There is not really any reason to use a `File` for this purpose, and (obviously) it adds a lot of confusion. Hence the impetus to disallow relative paths.